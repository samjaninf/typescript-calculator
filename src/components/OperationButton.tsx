import { BoundAction } from '../types'
import { StyleAttribute, css, hover } from 'glamor'
import { h } from 'preact'
import Button from './Button'

type OperationButtonProps = {
  action: BoundAction,
  className?: StyleAttribute,
  text: string,
}

const operationButton: StyleAttribute = css(
  { background: '#ff8754' },
  hover({ color: '#ff8754' }),
)

const OperationButton = ({ action, className, text }: OperationButtonProps): JSX.Element => (
  <Button
    action={action}
    className={css(operationButton, className)}
    text={text}
  />
)

export default OperationButton
