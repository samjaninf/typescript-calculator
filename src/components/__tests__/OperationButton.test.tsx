import OperationButton from '../OperationButton'
import snapshot from './utils/snapshot'

const action = () => ({ type: 'ADD' })

describe('OperationButton', () => {
  it('renders', () => snapshot(OperationButton))
  it.skip('passes action property', () => snapshot(OperationButton, { action }))
  it('passes className property', () => snapshot(OperationButton, { className: 'css-1j2u492' }))
  it('passes text property', () => snapshot(OperationButton, { text: '+' }))
})
