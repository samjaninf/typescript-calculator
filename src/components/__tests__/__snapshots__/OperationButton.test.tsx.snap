// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationButton passes className property 1`] = `"<Button class></Button>"`;

exports[`OperationButton passes text property 1`] = `"<Button class text=\\"+\\"></Button>"`;

exports[`OperationButton renders 1`] = `"<Button class></Button>"`;
